mlflow/utils/virtualenv.py

```diff
@@ -239,6 +240,22 @@ def _get_virtualenv_activate_cmd(env_dir: Path) -> str:
     return f"source {activate_cmd}" if not is_windows() else str(activate_cmd)
 
 
+class PermissionError(Exception):
+    """
+    Custom exception to handle permission errors when checking for the existence of a directory.
+    """
+
+
+def _check_dir_exists(dir: Path):
+    try:
+        return dir.exists()
+    except Exception as e:
```

Comment:
can we catch `PermissionError`? `Exception` is too broad.

Reply:
Yea seems like it throws the built-in PremissionError, updated